### PR TITLE
fix: worker init auth in catch, death loop restart

### PR DESCRIPTION
**Pull request by DDDHuang**

### CODE REVIEW

**Comment by DDDHuang** [comment]
Branch for test. 
And maybe we can retag 3.0.13

File: src/connection/mc_worker.erl [hunk 1]
```diff
@@ -215,6 +215,15 @@ get_set_opts_module(Options) ->
...
 auth_if_credentials(Socket, ConnState, NetModule, Login, Password) ->
-  Version = mc_worker_logic:get_version(Socket, ConnState#conn_state.auth_source, NetModule),
-  mc_auth_logic:auth(Version, Socket, ConnState#conn_state.auth_source, Login, Password, NetModule),
-  ok.
+  try
+    Version = mc_worker_logic:get_version(Socket, ConnState#conn_state.auth_source, NetModule),
+    true = mc_auth_logic:auth(Version, Socket, ConnState#conn_state.auth_source, Login, Password, NetModule),
+    ok
+  catch E:R:S ->
+    logger:error("Mongo client auth failed ~p: ~p: ~0p", [E, R, S]),
```
Comment (zmstone) [inline src/connection/mc_worker.erl:223]: is the stacktrace interesting at all ?

Comment (zmstone) [reply src/connection/mc_worker.erl:223]: in mc_auth_logic:scram_sha_1_auth for example, the stacktrace is already provided as error reason (`R`)

Comment (zmstone) [reply src/connection/mc_worker.erl:223]: please apply this:
```
diff --git a/src/connection/mc_auth_logic.erl b/src/connection/mc_auth_logic.erl
index a3fdac6..f3458d4 100644
--- a/src/connection/mc_auth_logic.erl
+++ b/src/connection/mc_auth_logic.erl
@@ -52,7 +52,7 @@ scram_sha_1_auth(Socket, Database, Login, Password, SetOpts) ->
     scram_first_step(Socket, Database, Login, Password, SetOpts)
   catch
     Error:Reason:StackTrace ->
-      erlang:error({<<"Can't pass authentification">>, Error, Reason, StackTrace})
+      erlang:raise(Error, {cannot_pass_auth, Reason}, StackTrace)
   end.
```

Comment (DDDHuang) [reply src/connection/mc_worker.erl:223]: Just discussed with xinyu about the order of starting the worker, may have to change the logic of some more starts, and to do a health check to report errors. So ignore me , and I will convert it to draft

**Comment by terry-xiaoyu** [comment]
fixed in https://github.com/emqx/mongodb-erlang/pull/30

**Pull request closed without merging**